lib/server.js

```diff
@@ -844,30 +870,48 @@ WalletService.prototype.createTx = function(opts, cb) {
           return cb(new ClientError('NOTALLOWEDTOCREATETX', 'Cannot create TX proposal during backoff time'));
 
         var copayer = wallet.getCopayer(self.copayerId);
-        var hash = WalletUtils.getProposalHash(opts.toAddress, opts.amount, opts.message, opts.payProUrl);
+        var proposalHeader = Model.TxProposal.create(opts).getHeader();
+        var hash = WalletUtils.getProposalHash.apply(WalletUtils, proposalHeader);
         if (!self._verifySignature(hash, opts.proposalSignature, copayer.requestPubKey))
           return cb(new ClientError('Invalid proposal signature'));
 
-        var toAddress;
-        try {
-          toAddress = new Bitcore.Address(opts.toAddress);
-        } catch (ex) {
+        var outputs = (opts.type == Model.TxProposal.Types.MULTIPLEOUTPUTS)
+          ? opts.outputs
+          : [ { toAddress: opts.toAddress, amount: opts.amount } ];
+        var badAddress = false,
```

Comment:
My original implementation of this block was this:

        var outputs = (opts.type == Model.TxProposal.Types.MULTIPLEOUTPUTS)
          ? opts.outputs
          : [ { toAddress: opts.toAddress, amount: opts.amount } ];
        _.each(outputs, function(output) {
          var toAddress;
          try {
            toAddress = new Bitcore.Address(output.toAddress);
          } catch (ex) {
            return cb(new ClientError('INVALIDADDRESS', 'Invalid address'));
          }
          if (toAddress.network != wallet.getNetworkName())
            return cb(new ClientError('INVALIDADDRESS', 'Incorrect address network'));
          if (output.amount <= 0)
            return cb(new ClientError('Invalid amount'));
          if (output.amount < Bitcore.Transaction.DUST_AMOUNT)
            return cb(new ClientError('DUSTAMOUNT', 'Amount below dust threshold'));
        });

But _.each() somehow executed the callbacks multiple times, and the mocha test fails. I assumed that there is a bug in lodash. To avoid that odd behavior, I used the flags instead.